zilliqa/src/consensus.rs

```diff
@@ -1153,7 +1153,7 @@ impl Consensus {
             proposal.header.transactions_root_hash,
             proposal.header.receipts_root_hash,
             proposal.transactions,
-            proposal.header.timestamp,
+            SystemTime::max(SystemTime::now(), parent_block.header.timestamp),
```

Comment:
This timestamp in the block should match the timestamp read by contracts in this transaction via `block.timestamp`. This is the timestamp we pass when executing transactions via `executed_block_header`.

Reply:
Thanks, I saw it passed in to tx processing but assumed tests would have caught if it broke.

Closing